### PR TITLE
Adding style transfer functionality into "image_processing_demo"

### PR DESCRIPTION
**Pull request by ieliz**
Adding functionality of the deleted "style_transfer_sample" into "image_processing_demo".
The sample could be found here:
https://github.com/openvinotoolkit/openvino/tree/releases/2021/3/inference-engine/samples/style_transfer_sample

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by akorobeinikov** [comment]
@ieliz Please update documentation for `image_processing` demo.

File: demos/common/cpp/models/src/style_transfer_model.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+using namespace InferenceEngine;
+
+StyleTransferModel::StyleTransferModel(const std::string& modelFileName) :
+    ImageModel(modelFileName, false) {
+}
+
+template<typename T>
+void tell_me_type(T t, std::string name) {
+    std::cout << name << " " << __PRETTY_FUNCTION__ << std::endl;
+}
```
Comment (akorobeinikov) [inline demos/common/cpp/models/src/style_transfer_model.cpp:35]: Where you use this function? If it don't used, please delete it.

Comment (ieliz) [reply demos/common/cpp/models/src/style_transfer_model.cpp]: Done

**Review by akorobeinikov — COMMENTED** [review]
Please add style model `fast-neural-style-mosaic-onnx` to file `models.lst`

**Review by akorobeinikov — APPROVED** [review]

**Pull request merged**